docker/scripts/start_musicbrainz_server.sh

```diff
@@ -21,6 +21,10 @@ if [ -s "$PID_FILE" ]; then
     fi
 fi
 
+chpst -u musicbrainz:musicbrainz \
+    carton exec -- ./script/dbdefs_exists \
+    || exit $?
+
```

Comment:
I was looking at some of the server logs and reading through consul-template docs to understand this a bit more: https://github.com/hashicorp/consul-template/blob/master/docs/modes.md#exec-mode

In exec mode, consul-template doesn't supervise the `start_server` process -- if it dies then consul-template dies with it.  In that case, runit should restart consul-template, which would be good.  But that doesn't seem to be what's happening; `start_server` is staying alive while plackup is failing:

```
[error] Caught exception in engine "There is no configuration in DBDefs for database READWRITE but one is required at lib/MusicBrainz/Server/DatabaseConnectionFactory.pm line 47.
MusicBrainz::Server::DatabaseConnectionFactory::get_connection("MusicBrainz::Server::DatabaseConnectionFactory", "READWRITE", "fresh", 0) called at lib/MusicBrainz/Server/Context.pm line 46
...
```

So I'm not sure this bit of code being added will actually run when we want, because `start_musicbrainz_server.sh` isn't being invoked multiple times.

Another issue, I think, is that DBDefs isn't actually non-existent or corrupt here -- as mentioned on IRC, just the `register_databases` call is receiving empty input.  The actual exception occurs in DatabaseConnectionFactory when it can't find READWRITE.  So script/dbdefs_exists won't actually indicate that anything is wrong, because the DBDefs module itself will load fine.

Reply:
Where does the above error message comes from? It was not in yesterday’s log of MB WS container on `trille`.

Reply:
> So we should rather have start_server to call database_exists before running plackup?

Well, start_server isn't our code, so I don't think this is possible.  But start_server isn't being invoked again -- the service itself never dies or restarts.

> Where does the above error message come from? It was not in yesterday’s log of MB WS container on trille.

It's from all website containers (from 8/6 on trille), so not the same day or the exact same issue, but same behavior.  The service didn't restart when it encountered "Can't use an undefined value as a subroutine reference" either, otherwise you'd see it from the logs.

Reply:
Sorry, I did not intend to change `start_server` itself but rather having a wrapper or something to make `start_server` die when “`register_databases` call is receiving empty input”.